packages/vant/src/tabbar-item/TabbarItem.tsx

```diff
@@ -59,11 +59,17 @@ export default defineComponent({
         const { $route } = vm;
         const { to } = props;
         const config = isObject(to) ? to : { path: to };
-        return !!$route.matched.find((val) => {
-          const pathMatched = 'path' in config && config.path === val.path;
-          const nameMatched = 'name' in config && config.name === val.name;
-          return pathMatched || nameMatched;
-        });
+        const pathMatched =
+          'path' in config &&
+          $route.matched.some((val) => config.path === val.path);
+        const nameMatched =
+          'name' in config &&
+          $route.matched.some((val) => config.name === val.name);
+        const isActive = pathMatched || nameMatched;
+        if (isActive) {
+          parent.setActive(props.name ?? index.value, () => {});
```

Comment:
Computed getter functions should only perform pure computation and be free of side effects.